Objects/structseq.c

```diff
@@ -32,18 +32,65 @@ PyObject *
 PyStructSequence_New(PyTypeObject *type)
 {
     PyStructSequence *obj;
-    Py_ssize_t size = REAL_SIZE_TP(type), i;
+    PyObject *key, *value;
+    Py_ssize_t size, visible_size, i;
+
+    if (Py_TYPE(type) != &PyType_Type) {
+        goto wrong_type;
+    }
+    if (type->tp_base != &PyTuple_Type) {
+        goto wrong_type;
+    }
+
+    key = _PyUnicode_FromId(&PyId_n_fields);
+    if (key == NULL) {
+        return NULL;
+    }
+
+    value = PyDict_GetItemWithError(type->tp_dict, key);
```

Comment:
You can use `_PyDict_GetItemIdWithError()`.

But first check that `tp_dict != NULL`.